doc/source/whatsnew/v0.25.0.rst

```diff
@@ -134,7 +134,7 @@ Interval
 Indexing
 ^^^^^^^^
 
--
+- Bug when a date string with a UTC offset would get ignored during indexing. (:issue:`24076`, :issue:`16785`)
```

Comment:
I think this is a signficant enough change that it needs a sub-section in api breaking (yes its a bug, but may lead to incorrect results, so better if its highly visible)